src/Microsoft.InsightsGenerator/SigGen.cs

```diff
@@ -8,8 +8,8 @@ public SignatureGenerator(DataArray table)
         {
             this.table = table;
         }
-        
-        public Learn()
+
+        public SignatureGeneratorResult Learn()
```

Comment:
qq, we would just call a bunch of methods from this function and add insights to the results right?

Reply:
Yeah, we will create one method for each kind of insight(Avg, mean, top3, etc.) and call them from the learn method. 
These methods will keep adding insights to result.insight Dictionary.